pallets/runtime/src/bridge.rs

```diff
@@ -330,20 +351,9 @@ decl_module! {
                 ensure!(Self::frozen_txs(&bridge_tx), Error::<T>::NoSuchFrozenTx);
                 <FrozenTxs<T>>::remove(&bridge_tx);
                 Self::deposit_event(RawEvent::UnfrozenTx(bridge_tx.clone()));
-                if let Some(PendingTx {
-                        did,
-                        bridge_tx,
-                }) = Self::issue(bridge_tx.clone())? {
-                    <PendingTxs<T>>::mutate(did, |pending_txs| {
-                        pending_txs.push(bridge_tx.clone())
-                    });
-                    Self::deposit_event(RawEvent::Pending(PendingTx {
-                        did,
-                        bridge_tx
-                    }));
-                } else {
-                    <HandledTxs<T>>::insert(&bridge_tx, true);
-                    Self::deposit_event(RawEvent::Bridged(bridge_tx));
+                ensure!(!Self::handled_txs(&bridge_tx), Error::<T>::ProposalAlreadyHandled);
```

Comment:
Duplicate check for `handled_txs`. BTW we can merge `handled_txs` & `freeze_txs` into struct some thing like that
```
struct BridgeTxStatus {
 pub handled_tx: bool,
pub freeze_tx: bool
}

TxStatus get(tx_status): map BridgeTx<T::AccountId, T::Balance> => BridgeTxStatus;
```

Reply:
Thanks, a merge artefact again! About `BridgeTxStatus`, I totally agree with you but I think it must be done as a separate task after we finish with adding this functionality. The reason being it's a bit more complex then that since there are more states: also Pending and Timelocked.